src/Splat.Microsoft.Extensions.DependencyInjection/MicrosoftDependencyResolver.cs

```diff
@@ -91,29 +91,27 @@ public virtual IEnumerable<object> GetServices(Type? serviceType, string? contra
         var isNull = serviceType is null;
         serviceType ??= typeof(NullServiceType);
 
-        IEnumerable<object> services;
+        IEnumerable<object> services = Enumerable.Empty<object>();
 
         if (contract is null || string.IsNullOrWhiteSpace(contract))
         {
             // this is to deal with CS8613 that GetServices returns IEnumerable<object?>?
             services = ServiceProvider.GetServices(serviceType)
                 .Where(a => a is not null)
                 .Select(a => a!);
-
-            if (isNull)
-            {
-                services = services
-                    .Cast<NullServiceType>()
-                    .Select(nst => nst.Factory()!);
-            }
         }
-        else
+        else if (ServiceProvider is IKeyedServiceProvider serviceProvider)
```

Comment:
@glennawatson as of net8 ServiceProvider is IKeyedServiceProvider, keep code as-is until we drop net6? then change the ctor on L36?

Reply:
Perhaps we could set a milestone for November to remind us to review this.